src/server/routes/contract/events/getEventLogs.ts

```diff
@@ -0,0 +1,142 @@
+import { Static, Type } from "@sinclair/typebox";
+import { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import {
+  getEventLogsByBlockTimestamp,
+  getEventLogsByCreationTimestamp,
+} from "../../../../db/contractEventLogs/getContractEventLogs";
+import { createCustomError } from "../../../middleware/error";
+import { standardResponseSchema } from "../../../schemas/sharedApiSchemas";
+
+const requestQuerySchema = Type.Union([
+  Type.Object({
+    contractAddresses: Type.Optional(Type.Array(Type.String())),
+    topics: Type.Optional(Type.Array(Type.String())),
+    fromCreationTimestamp: Type.Number(),
+    toCreationTimestamp: Type.Optional(Type.Number()),
+  }),
+  Type.Object({
+    contractAddresses: Type.Optional(Type.Array(Type.String())),
+    topics: Type.Optional(Type.Array(Type.String())),
+    fromBlockTimestamp: Type.Number(),
```

Comment:
should this be `fromBlockNumber`??

Reply:
No, block number is chain dependent. If I want to get all the events on my contracts across different chains that happened within a certain period of time, I need fromBlockTimestamp, toBlockTimestamp. 

Also, if I am querying a moving window in my application:
1. set lastIndexedTimestamp = new Date.getTime()
2. set currentTimestamp = new Date.getTime()
3. query { lastIndexedTimestamp, currentTimestamp }
4. set lastIndexedTimestamp = currentTimestamp
5. jmp 2

then I cannot use fromBlockTimestamp, I need to use fromCreationTimestamp. This is because the indexer will have a slight delay (that will vary by workload), causing the indexer to add blockTimestamp at blockTimestamp + delay. Using creation time will negate the need for adding a look back to the sliding window especially bc it could vary.

Reply:
> If I want to get all the events on my contracts across different chains that happened within a certain period of time, I need fromBlockTimestamp, toBlockTimestamp.

Then can't we achieve the same with `fromCreationTimestamp`/`toCreationTimestamp`??